### PR TITLE
Hierarchical Agentic RAG example

### PR DESCRIPTION
**Pull request by minmin-intel**
## Description

Add an example of hierarchical multi-agent system to improve RAG performance

## Issues

[List the issue or RFC link this PR is working on. If there is no such link, please mark it as `n/a`.](https://github.com/opea-project/GenAIExamples/issues/414)

## Type of change

List the type of change like below. Please delete options that are not relevant.
- [x] New feature (non-breaking change which adds new functionality)

## Dependencies

1. https://github.com/opea-project/GenAIComps/pull/480

2. OpenAI model access

## Tests

Tested locally with the script in tests/ folder and passed tests.

### CODE REVIEW

File: HierarchicalAgenticRAG/README.md [hunk 1]
```diff
@@ -0,0 +1,106 @@
...
+```
+cd GenAIComps
+docker build -t opea/comps-agent-langchain:latest --build-arg https_proxy=$https_proxy --build-arg http_proxy=$http_proxy -f comps/agent/langchain/docker/Dockerfile .
+```
+
+2. Launch tool services </br>
+   In this example, we will use some of the mock APIs provided in the Meta CRAG KDD Challenge to demonstrate the benefits of gaining additional context from mock knowledge graphs.
+
+```
+docker run -d -p=8080:8000 docker.io/aicrowd/kdd-cup-24-crag-mock-api:v0
```
Comment (xuechendi) [inline HierarchicalAgenticRAG/README.md:46]: is this example only works with kdd-cup-crag? Can we use this docker for other dataset?

File: HierarchicalAgenticRAG/tools/worker_agent_tools.yaml [hunk 2]
```diff
@@ -0,0 +1,5 @@
+# Copyright (C) 2024 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+duckduckgo_search:
+  callable_api: ddg-search
```
Comment (xuechendi) [inline HierarchicalAgenticRAG/tools/worker_agent_tools.yaml:5]: duckduckgo is just a placeholder for RagRetriever right? 

Does that mean we still need to have DocIndexer to complete the PR eventually?
https://github.com/opea-project/GenAIExamples/issues/415

File: HierarchicalAgenticRAG/README.md (hunk 1 above)
Comment (minmin-intel) [reply HierarchicalAgenticRAG/README.md:46]: kdd-cup-crag APIs are used as example tools. users can define their own tools by changing the yaml and tools.py in tools folder.

File: HierarchicalAgenticRAG/tools/worker_agent_tools.yaml (hunk 2 above)
Comment (minmin-intel) [reply HierarchicalAgenticRAG/tools/worker_agent_tools.yaml:5]: Yes, we will update this example in v1.0 or later with retrieval megaservice as mentioned in the roadmap section in readme.

**Review by xuechendi — APPROVED** [review]
LGTM

**Comment by minmin-intel** [comment]
@XuhuiRen I have updated the example name to AgentQnA and also updated readme.

**Comment by minmin-intel** [comment]
UT test_agentqna.sh with OpenAI GPT-4o-mini passed. See screenshots below. @chensuyue 
image
image
image

File: AgentQnA/assets/hierarchical_agentic_rag_arch.png
Comment (XuhuiRen) [inline AgentQnA/assets/hierarchical_agentic_rag_arch.png:1]: should be better change the file name

Comment (minmin-intel) [reply AgentQnA/assets/hierarchical_agentic_rag_arch.png:1]: @XuhuiRen  I changed the picture filename.

**Review by XuhuiRen — APPROVED** [review]

**Pull request merged**